.github/workflows/unit_test_pipeline.yml

```diff
@@ -0,0 +1,66 @@
+name: TM1Py Integration Tests
+
+on:
+  workflow_dispatch:
+    inputs:
+      environments:
+        description: 'JSON array of environments to test (e.g., ["tm1-11-onprem", "tm1-11-cloud"])'
+        required: true
+        default: '["tm1-11-onprem","tm1-11-cloud","tm1-12-mcsp"]'
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        environment: ${{ fromJson(github.event.inputs.environments) }}
+    environment: ${{ matrix.environment }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.x'
+
+      - name: Install dependencies
+        run: |
+          pip install -e .[unit_test]
+
+      - name: Retrieve TM1 Connection Details
+        id: get-connection
+        run: echo "Retrieving TM1 connection details"
+        env:
+          TM1_CONNECTION: ${{ vars.TM1_CONNECTION }}
+          TM1_CONNECTION_SECRET: ${{ secrets.TM1_CONNECTION_SECRET }}
```

Comment:
Are these vars environment specific?

Reply:
Yes, they are. I think the above chart shows it.

Reply:
Perfect 😃